Quick_Deploy/vLLM/README.md

```diff
@@ -0,0 +1,127 @@
+```
+
+This step might take a while to build the environment packages. Once complete, the provided sample [model_repository](model_repository) will be populated with `triton_python_backend_stub` and `vllm_env.tar.gz`.
+
+## Step 2: Set Up Triton Inference Server
+
+The structure of the sample model repository as follows after the Step 1 above:
+```
+model_repository/
+`-- vllm
+    |-- 1
+    |   `-- model.py
+    |-- config.pbtxt
+    |-- triton_python_backend_stub
+    |-- vllm_engine_args.json
+    `-- vllm_env.tar.gz
+
+```
+
+A sample model repository for deploying `facebook/opt-125m` using vLLM in Triton is included with this demo as `model_repository` directory. The content of `vllm_engine_args.json` is:
+
+```json
+{
+    "model": "facebook/opt-125m",
+    "disable_log_requests": "true"
+}
+```
+This file can be modified to provide further settings to the vLLM engine. See vLLM [AsyncEngineArgs](https://github.com/vllm-project/vllm/blob/32b6816e556f69f1672085a6267e8516bcb8e622/vllm/engine/arg_utils.py#L165) and [EngineArgs](https://github.com/vllm-project/vllm/blob/32b6816e556f69f1672085a6267e8516bcb8e622/vllm/engine/arg_utils.py#L11) for supported key-value pairs.
+
+*Note*: vLLM greedily consume upto 90% of the GPU's memory under default settings. You can provide appropriate fields like `gpu_memory_utilization` and other settings via [`vllm_engine_args.json`](model_repository/vllm/vllm_engine_args.json).
+
+Read through the documentation in [`model.py`](model_repository/vllm/1/model.py) to understand how to configure this sample for your use-case.
+
+Start the server like below:
+
+```
+docker run -p 8000:8000 -p 8001:8001 -p 8002:8002 --privileged --gpus all -it --rm --shm-size=8G --ulimit memlock=-1 --ulimit stack=67108864 -v ${PWD}/model_repository:/models nvcr.io/nvidia/tritonserver:23.08-py3 tritonserver --model-store=/models
+```
+
+Upon successful start of the server, you can see the following in the end.
+
+```
+I0901 23:39:08.729123 1 grpc_server.cc:2451] Started GRPCInferenceService at 0.0.0.0:8001
+I0901 23:39:08.729640 1 http_server.cc:3558] Started HTTPService at 0.0.0.0:8000
+I0901 23:39:08.772522 1 http_server.cc:187] Started Metrics Service at 0.0.0.0:8002
+```
+
+## Step 3: Using a Triton Client to Query the Server
+
+We will run the client within Triton's SDK container to issue multiple async requests using
+[gRPC asyncio client](https://github.com/triton-inference-server/client/blob/main/src/python/library/tritonclient/grpc/aio/__init__.py)
+library.
+
+```
+docker run -it --net=host -v ${PWD}:/workspace/ nvcr.io/nvidia/tritonserver:23.08-py3-sdk bash
+```
+
+Within the container, run the [`client.py`](client.py) as:
+
+```
+python3 client.py
+```
+
+The client reads prompts from [prompts.txt](prompts.txt) file, sends them to Triton server for inference and stores the results into a file named `results.txt` by default.
```

Comment:
This could be a "version 2" type thing, but personally if I was the user experimenting with the vLLM example, I'd rather some interactive and intuitive interface like this rather than reading from a file:
```bash
python client.py --prompt "Hello, my name is"
```

On this note, it could also be a good place to soft demo the concept of OpenAI format to entice some users by changing the UX to something like:
```bash
python3 client.py -d '{"prompt": "San Francisco is a", "use_beam_search": true, "n": 4, "temperature": 0 }'
```
and internally the example shows how to convert the json dict into Triton inputs as a reference point for users writing their own clients.